src/import.c

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "storage.h"
+#include "util.h"
+#include "module.h"
+#include "import.h"
+#include "entry.h"
+#include <string.h>
+#include <stdio.h>
+
+
+/**
+ * CacheImport:
+ * @shadow_path: Path to shadow file.
+ * @shadow_hash: Hashed password. 
+ * @storage_path: Path to store policy cache entry.
+ * @username: Username to create cache entry for.
+ * @error: Errors.
+ *
+ * Returns: #True if CacheEntry was created or #False on failure.
+ */
+
+gboolean CacheImport(const char *shadow_path, const char *storage_path, const char *username, GError **error) {
+  if (!shadow_path) {
+    shadow_path = "/etc/shadow";
+  }
+  
+  if (!storage_path) {
+    storage_path = DEFAULT_STORAGE_PATH;
+  }
+
+  GError *errors = NULL;
+  // Get user's hashed password.
+  GBytes* shadow_hash = CacheUtilReadShadowFile(shadow_path, username, &errors);
+
+  // Create a new Cache Entry.
+  CacheEntry *entry = CacheEntryNew();
+
+  // Set the newly created entry hash using the /etc/shadow hash.
+  CacheEntryHashSet(entry, CACHE_ENTRY_ALGORITHM_CRYPT, shadow_hash);
+
+  GError *tmp_error = NULL;
```

Comment:
Remove this tmp_error usage too. It obscures the actual error. Just pass error to CacheStoragePutEntry.

If you really want to add something more to the error message, you can do g_prefix_error.